### PR TITLE
Migrate to Java 21

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/websub/codeaction/AbstractCodeActionTest.java [hunk 1]
```diff
@@ -55,13 +53,13 @@
...
     private static final Gson GSON = new Gson();
 
-    @Test(dataProvider = "testDataProvider")
-    public void testCodeActions(String srcFile, int line, int offset, String resultFile)
-            throws IOException {
-        Path srcPath = getResourcePath("ballerina_sources", getTestPackage(), srcFile);
-        Path targetPath = getResourcePath("codeaction", getConfigDir(), resultFile);
-        performTest(srcPath, LinePosition.from(line, offset), targetPath);
-    }
+//    @Test(dataProvider = "testDataProvider")
```
Comment (warunalakshitha) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/websub/codeaction/AbstractCodeActionTest.java:56]: Shall we create a issue for this?

Comment (warunalakshitha) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/websub/codeaction/AbstractCodeActionTest.java:56]: May be we can reenable and see. If fail , we can disable (without commenting ) and create an issue

Comment (HindujaB) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/websub/codeaction/AbstractCodeActionTest.java]: It seems it is still failing with an OOM error.

Comment (HindujaB) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/websub/codeaction/AbstractCodeActionTest.java:56]: Disabled the test and created https://github.com/ballerina-platform/ballerina-library/issues/7373

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**